File: app/applet/generate_icons.py
from PIL import Image
import cairosvg
import io

def generate_png(svg_path, png_path, size):
    # Convert SVG to PNG using cairosvg
    png_data = cairosvg.svg2png(url=svg_path, output_width=size, output_height=size)
    
    # Save PNG
    with open(png_path, 'wb') as f:
        f.write(png_data)
    
    print(f'Generated {png_path}')

generate_png('public/logo.svg', 'public/pwa-192x192.png', 192)
generate_png('public/logo.svg', 'public/pwa-512x512.png', 512)
generate_png('public/logo.svg', 'public/pwa-512x512-maskable.png', 512)

File: generate-icons.js
import sharp from 'sharp';
import fs from 'fs';
import path from 'path';

const svgBuffer = fs.readFileSync(path.resolve('./public/logo.svg'));

async function generate() {
  await sharp(svgBuffer).resize(192, 192).png().toFile('./public/pwa-192x192.png');
  await sharp(svgBuffer).resize(512, 512).png().toFile('./public/pwa-512x512.png');
  await sharp(svgBuffer).resize(512, 512).png().toFile('./public/pwa-512x512-maskable.png');
  console.log('Icons generated');
}

generate().catch(console.error);
